controller/aks-cluster-config-handler.go

```diff
@@ -0,0 +1,686 @@
+
+	// set tags
+	upstreamSpec.Tags = make(map[string]string)
+	if len(clusterState.Tags) != 0 {
+		upstreamSpec.Tags = to.StringMap(clusterState.Tags)
+	}
+
+	// set agentpool profile
+	for _, ng := range *clusterState.AgentPoolProfiles {
+		var upstreamNG aksv1.NodeGroup
+		upstreamNG.Name = to.String(ng.Name)
+		upstreamNG.Count = to.Int32(ng.Count)
+		upstreamNG.MaxPods = to.Int32(ng.MaxPods)
+		upstreamNG.VMSize = string(ng.VMSize)
+		upstreamNG.OsDiskSizeGB = to.Int32(ng.OsDiskSizeGB)
+		upstreamNG.OsDiskType = string(ng.OsDiskType)
+		upstreamNG.Mode = string(ng.Mode)
+		upstreamNG.OsType = string(ng.OsType)
+		upstreamNG.Version = to.String(ng.OrchestratorVersion)
+		upstreamNG.AvailableZones = to.StringSlice(ng.AvailabilityZones)
+		upstreamSpec.NodeGroups = append(upstreamSpec.NodeGroups, upstreamNG)
+	}
+
+	// set network configuration
+	networkProfile := clusterState.NetworkProfile
+	if networkProfile != nil {
+		upstreamSpec.NetworkPlugin = string(networkProfile.NetworkPlugin)
+		upstreamSpec.NetworkDNSServiceIP = to.String(networkProfile.DNSServiceIP)
+		upstreamSpec.NetworkDockerBridgeCIDR = to.String(networkProfile.DockerBridgeCidr)
+		upstreamSpec.NetworkServiceCIDR = to.String(networkProfile.ServiceCidr)
+		upstreamSpec.NetworkPolicy = string(networkProfile.NetworkPolicy)
+		upstreamSpec.NetworkPodCIDR = to.String(networkProfile.PodCidr)
+		upstreamSpec.LoadBalancerSku = string(networkProfile.LoadBalancerSku)
+	}
+
+	// set linux account profile
+	linuxProfile := clusterState.LinuxProfile
+	if linuxProfile != nil {
+		upstreamSpec.LinuxAdminUsername = to.String(linuxProfile.AdminUsername)
+		sshKeys := *linuxProfile.SSH.PublicKeys
+		upstreamSpec.LinuxSSHPublicKeyContents = to.String(sshKeys[0].KeyData)
+	}
+	return upstreamSpec, err
+}
+
+func updateUpstreamClusterState(ctx context.Context, config *aksv1.AKSClusterConfig, upstreamSpec *aksv1.AKSClusterConfigSpec) (*aksv1.AKSClusterConfig, error) {
+	// check kubernetes version for update
+
+	if config.Spec.Tags == nil {
+		config.Spec.Tags = map[string]string{}
+	}
+	config.Spec.Tags["displayName"] = config.Spec.DisplayName
+	if reflect.DeepEqual(config.Spec.Tags, upstreamSpec.Tags) {
+		logrus.Debugf("[AKS] updateUpstreamClusterState check config are similar, exit \n")
+		return config, nil
+	}
+
+	azureAuthorizer, _ := newClientAuthorizer(config.Spec)
+	resourceGroupsClient, _ := newResourceGroupsClient(azureAuthorizer, config.Spec)
+
+	exists, _ := checkResourceGroup(ctx, resourceGroupsClient, config.Spec)
+	if !exists {
+		logrus.Infof("Resource group %s does not exist, creating", config.Spec.ResourceGroup)
+		status, err := createResourceGroup(ctx, resourceGroupsClient, config.Spec)
+		if status {
+			logrus.Infof("Resource grup [%s] updated successfully", config.Name)
+		}
+		if err != nil {
+			fmt.Errorf("Error during creating resource group %v", err)
+		}
+	}
+
+	resourceClusterClient, _ := newClustersClient(azureAuthorizer, config.Spec)
```

Comment:
Are we ignoring this error?

Reply:
this was fixed

Reply:
actually the err part was fixed but not the nil check on the client return